Add ruleset-optimization option in firewall settings.

phpb-com opened the pull request:
This setting will help with de-duplication of rules and reordering for better performance. From man pages:
<pre>
     set ruleset-optimization
           none      Disable the ruleset optimizer.
           basic     Enable basic ruleset optimization.  This is the default
                     behaviour.  Basic ruleset optimization does four things
                     to improve the performance of ruleset evaluations:

                     1.   remove duplicate rules
                     2.   remove rules that are a subset of another rule
                     3.   combine multiple rules into a table when
                          advantageous
                     4.   re-order the rules to improve evaluation performance

           profile   Uses the currently loaded ruleset as a feedback profile
                     to tailor the ordering of quick rules to actual network
                     traffic.

</pre>

phpb-com commented:
I'll submit a patch to add GUI option for that today, consider this PR as WIP until then.

phpb-com commented:
Alright, this one commit should be all that is needed to add ruleset optimization. Since basic is the default, it is the one that is selected and added without user interaction. Please consider this PR for the merge.

AdSchellevis commented on src/etc/inc/filter.inc:
```diff
@@ -446,6 +446,12 @@ function filter_configure_sync($verbose = false)
         $limitrules .= "set limit table-entries {$config['system']['maximumtableentries']}\n";
     }
 
+    if ($config['system']['rulesoptimization'] <> "") {
```
@phpb-com better to check for empty($config['system']['rulesoptimization']) here, rulesoptimization is optional available and may not be initialised.

phpb-com replied:
Thanks, let me rework the diff and submit a new one. I think that I also better rename it to rulesetoptimization for consistency.

AdSchellevis replied:
@phpb-com sounds good, thanks

phpb-com commented:
Ok, I've updated the diff and changed how the check for config presence is done, as well as the name of the configuration. Thanks for the comments @AdSchellevis

AdSchellevis commented:
@phpb-com thanks for fixing!
